dev/install-common-deps.sh

```diff
@@ -58,6 +58,9 @@ if [[ "$INSTALL_LARGE_PYTHON_DEPS" == "true" ]]; then
   ls -lha $(find $CONDA_DIR/envs/test-environment/ -path "*bin/spark-*")
 fi
 
+# Install `mlflow-test-plugin` without dependencies
+pip install --no-dependencies tests/resources/mlflow-test-plugin
```

Comment:
With this approach, we can:

- avoid installing the latest version of MLflow on PyPI and its dependencies
- avoid tweaking `tests/resources/mlflow-test-plugin/setup.py`
- test `mlflow-test-plugin` using the dev version of MLflow